operator/builtin/transformer/flatten/flaltten.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flatten
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/errors"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("flatten", func() operator.Builder { return NewFlattenOperatorConfig("") })
+}
+
+// NewFlattenOperatorConfig creates a new flatten operator config with default values
+func NewFlattenOperatorConfig(operatorID string) *FlattenOperatorConfig {
+	return &FlattenOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "flatten"),
+	}
+}
+
+// FlattenOperatorConfig is the configuration of a flatten operator
+type FlattenOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+	Field                    entry.BodyField `mapstructure:"field" json:"field" yaml:"field"`
+}
+
+// Build will build a Flatten operator from the supplied configuration
+func (c FlattenOperatorConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(c.Field.String(), "$attributes") || strings.Contains(c.Field.String(), "$resource") {
```

Comment:
So, I wanted to note why this works the way it does. When it is unmarshalled it is allows the prefix for "$attributes" and "$resource" through and just saves it as a key. so since it doesn't it catch it during unmarshalling I'm catching it here, Unless we want to change how the bodyfield unmarshalls

Reply:
This is good